models/uniswap/ethereum/uniswap_v3_ethereum_proposals.sql

```diff
@@ -0,0 +1,73 @@
+{{ config(
+    schema = 'uniswap_v3_ethereum',
+    alias = 'proposals',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_time', 'blockchain', 'project', 'version', 'tx_hash'],
+    post_hook='{{ expose_spells(\'["ethereum"]\',
+                                "project",
+                                "uniswap_v3",
+                                \'["soispoke"]\') }}'
+    )
+}}
+
+{% set blockchain = 'ethereum' %}
+{% set project = 'uniswap' %}
+{% set project_version = 'v3' %}
+{% set dao_name = 'DAO: Uniswap' %}
+{% set dao_address = '0x408ed6354d4973f66138c91495f2f2fcbd8724c3' %}
+
+with cte_support as (SELECT 
+        voter as voter,
+        CASE WHEN support = 0 THEN sum(votes/1e18) ELSE 0 END AS votes_against,
+        CASE WHEN support = 1 THEN sum(votes/1e18) ELSE 0 END AS votes_for,
+        CASE WHEN support = 2 THEN sum(votes/1e18) ELSE 0 END AS votes_abstain,
+        proposalId
+FROM {{ source('uniswap_v3_ethereum', 'GovernorBravoDelegate_evt_VoteCast') }}
+GROUP BY support, proposalId, voter),
+
+cte_sum_votes as (
```

Comment:
How will this work for active proposals where vote counts are rolling? Since the unique key refers to proposal creation exclusively (thus 1 row = 1 proposal), it seems like this would update the row while the prop is rolling with every run of the incremental job, and if a proposal lives for longer than a week (not sure if this edge case is reasonable) we may have inconsistencies?

I'm thinking perhaps it makes more sense to keep both datasets solely focused on what they represent, and have a `view` on top that calculates latest status of votes.

Reply:
thanks, definitely a good catch you're right we have to be pretty careful about this. Tbh one straightforward option would just be to remove the incremental clause and refresh the spells on each run since the tables are really small?